lib/msf/core/exploit/pgadmin.rb

```diff
@@ -35,15 +47,30 @@ def check_version(patched_version, low_bound = 0)
     def csrf_token
       return @csrf_token if @csrf_token
 
-      res = send_request_cgi('uri' => normalize_uri(target_uri.path, 'login'), 'keep_cookies' => true)
-      set_csrf_token_from_login_page(res)
-      fail_with(Msf::Exploit::Failure::UnexpectedReply, 'Failed to obtain the CSRF token') unless @csrf_token
+      if auth_required?
+        res = send_request_cgi('uri' => normalize_uri(target_uri.path, 'login'), 'keep_cookies' => true)
+        set_csrf_token_from_login_page(res)
+      else
+        res = send_request_cgi('uri' => normalize_uri(target_uri.path, 'browser/js/utils.js'), 'keep_cookies' => true)
+        set_csrf_token_from_config(res)
+      end
+      fail_with(Failure::UnexpectedReply, 'Failed to obtain the CSRF token') unless @csrf_token
       @csrf_token
     end
 
+    def set_csrf_token_from_config(res)
```

Comment:
It would be nice to provide examples of what the csrf token looks like, in comments.